sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -319,6 +319,19 @@ def test_lasso_alpha_warning():
     assert_warns(UserWarning, clf.fit, X, Y)
 
 
+def test_lasso_fit_intercept():
+    X = [[-1], [0], [1]]
+    Y = [-1, 0, 1]
+
+    clf = Lasso(fit_intercept=False)
+    clf.fit(X, Y)
+    assert_equal(clf.coef_.shape, (1,))
+
+    clf2 = Lasso(fit_intercept=True)
+    clf2.fit(X, Y)
+    assert_equal(clf.coef_.shape, (1,))
```

Comment:
this test fails on master?

Reply:
I suggest this test instead, because `Lasso` is inherited from `ElasticNet` :

```python
def test_elastic_net_no_intercept_coef_shape():
    # {ElasticNet|Lasso}.fit(..) doesn't return coef_ with correct shape
    # when object initialized with fit_intercept = False

    X = [[-1], [0], [1]]
    y = [-1, 0, 1]

    for intercept in [True, False]:
        clf = ElasticNet(fit_intercept=intercept)
        clf.fit(X, y)
        coef_ = clf.coef_
        assert_equal(coef_.shape, (1,))
```